File: _10_Find_Index_of_First_Occurrence.java
public class _10_Find_Index_of_First_Occurrence {
    public static void main(String[] args) {
        Solution sol = new Solution();

        String haystack1 = "sadbutsad";
        String needle1 = "sad";
        System.out.println("Resultado 1: " + sol.strStr(haystack1, needle1)); 

        String haystack2 = "leetcode";
        String needle2 = "leeto";
        System.out.println("Resultado 2: " + sol.strStr(haystack2, needle2)); 
    }
}

// My solution
class Solution {
    public int strStr(String haystack, String needle) {
        // Solución rápida con método de Java
        return haystack.indexOf(needle);

        // --- Opción manual sin usar indexOf ---
        /*
        int n = haystack.length();
        int m = needle.length();

        for (int i = 0; i <= n - m; i++) {
            if (haystack.substring(i, i + m).equals(needle)) {
                return i;
            }
        }
        return -1;
        */
    }
}

File: _10_Find_Index_of_First_Occurrence.ts
function strStr(haystack: string, needle: string): number {
    return haystack.indexOf(needle);
}

const haystack1 = "sadbutsad";
const needle1 = "sad";
console.log(`Resultado 1: ${strStr(haystack1, needle1)}`); 

const haystack2 = "leetcode";
const needle2 = "leeto";
console.log(`Resultado 2: ${strStr(haystack2, needle2)}`); 

const haystack3 = "hello";
const needle3 = "ll";
console.log(`Resultado 3: ${strStr(haystack3, needle3)}`); 

const haystack4 = "a";
const needle4 = "a";
console.log(`Resultado 4: ${strStr(haystack4, needle4)}`); 

const haystack5 = "abc";
const needle5 = "c";
console.log(`Resultado 5: ${strStr(haystack5, needle5)}`); 
